src/components/experiment-tracking/parallel-coordinates/parallel-coordinates.js

```diff
@@ -76,11 +83,81 @@ export const ParallelCoordinates = ({ metricsData, selectedRuns }) => {
   };
 
   const handleMouseOverMetric = (e, key) => {
+    const runsCount = graph.find((each) => each[0] === key)[1].length;
     setHoveredAxisG(key);
+
+    const rect = e.target.getBoundingClientRect();
+
+    let x, direction;
+
+    if (window.innerWidth - rect.x > tooltipMaxWidth) {
+      x = e.clientX - sideBarWidth;
+      direction = 'right';
+    } else {
+      x = e.clientX - sideBarWidth - sideBarWidth / 2;
+      direction = 'left';
+    }
+    const y = rect.y - 140;
+
+    setShowTooltip({
+      content: {
+        label1: 'Metrics Name',
+        value1: key,
+        label2: 'Runs Count',
+        value2: runsCount,
+      },
+      direction,
+      pos: { x, y },
+      visible: true,
+    });
   };
 
   const handleMouseOutMetric = () => {
     setHoveredAxisG(null);
+    setShowTooltip({
+      pos: { x: -500, y: -500 },
+      visible: false,
+    });
+  };
+
+  const handleMouseOverLine = (e, key) => {
+    setHoveredElementId(key);
+
+    if (e) {
+      let x, direction;
+
+      if (window.innerWidth - e.clientX > tooltipMaxWidth) {
+        x = e.clientX - sideBarWidth;
+        direction = 'right';
+      } else {
+        x = e.clientX - sideBarWidth - sideBarWidth / 2;
+        direction = 'left';
+      }
+      const y = e.clientY - 150;
+
+      const parsedDate = new Date(formatTimestamp(key));
```

Comment:
```suggestion
      const y = e.clientY - 150;
      const parsedDate = new Date(formatTimestamp(key));
      let x, direction;

      if (window.innerWidth - e.clientX > tooltipMaxWidth) {
        x = e.clientX - sideBarWidth;
        direction = 'right';
      } else {
        x = e.clientX - sideBarWidth - sideBarWidth / 2;
        direction = 'left';
      }
```

We should try and declare variables at the top-level scope where possible.

And again, the `150` value here seems arbitrary: `const y = e.clientY - 150;`. It's fine if it is, maybe just leave a comment saying why it needs to be this value.

Reply:
yeh I'll leave a comment to explain the value, thanks for the suggestion